ext/strings_test.go

```diff
@@ -329,3 +331,398 @@ func TestStringsWithExtension(t *testing.T) {
+		if err != nil {
+			if expectedErr != "" {
+				if err.Error() != expectedErr {
+					t.Fatalf("expected %q as error message, got %q", expectedErr, err.Error())
+				}
+			} else {
+				t.Fatalf("unexpected error: %s", err)
+			}
+		} else {
+			if output.Type() != types.StringType {
+				t.Fatalf("expected test expr to eval to string (got %s instead)", output.Type().TypeName())
+			} else {
+				outputStr := output.Value().(string)
+				if outputStr != expectedOutput {
+					t.Errorf("expected %q as output, got %q", expectedOutput, outputStr)
+				}
+			}
+		}
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			out, err := runCase(tt.format, tt.formatArgs, tt.locale, t)
+			checkCase(out, tt.expectedOutput, err, tt.err, t)
+			if tt.locale == "" {
+				// if the test has no locale specified, then that means it
+				// should have the same output regardless of lcoale
+				t.Run("no change on locale", func(t *testing.T) {
+					out, err := runCase(tt.format, tt.formatArgs, "da_DK", t)
+					checkCase(out, tt.expectedOutput, err, tt.err, t)
+				})
+			}
+		})
+	}
+}
+
+func TestBadLocale(t *testing.T) {
+	_, err := cel.NewEnv(cel.Locale("bad-locale"), Strings())
+	if err != nil {
+		if err.Error() != "failed to parse locale: language: subtag \"locale\" is well-formed but unknown" {
+			t.Errorf("expected error messaged to be \"failed to parse locale: language: subtag \"locale\" is well-formed but unknown\", got %q", err)
+		}
+	} else {
+		t.Error("expected NewEnv to fail during locale parsing")
+	}
+}
+
+func TestLiteralOutput(t *testing.T) {
+	tests := []struct {
+		name          string
+		formatLiteral string
+		expectedType  string
+	}{
+		{
+			name:          "map literal support",
+			formatLiteral: `{"key1": b"xyz", "key2": duration("2h"), "key4": true, "key3": 2.71828}`,
+			expectedType:  `map`,
+		},
+		{
+			name:          "list literal support",
+			formatLiteral: `["abc", 3.14, [9, 8, 7, 6], timestamp("2023-02-03T23:31:20Z")]`,
+			expectedType:  `list`,
+		},
+	}
+	for _, tt := range tests {
+		parseAndEval := func(expr string, t *testing.T) (ref.Val, error) {
+			env, err := cel.NewEnv(Strings())
+			if err != nil {
+				t.Fatalf("cel.NewEnv(Strings()) failed: %v", err)
+			}
+			parsedAst, issues := env.Parse(expr)
+			if issues.Err() != nil {
+				t.Fatalf("env.Parse(%v) failed: %v", expr, issues.Err())
+			}
+			checkedAst, issues := env.Check(parsedAst)
+			if issues.Err() != nil {
+				t.Fatalf("env.Check(%v) failed: %v", expr, issues.Err())
+			}
+			program, err := env.Program(checkedAst)
+			if err != nil {
+				t.Fatal(err)
+			}
+			out, _, err := program.Eval(cel.NoVars())
+			return out, err
+		}
+		t.Run(tt.name, func(t *testing.T) {
+			expr := fmt.Sprintf(`"%%s".format([%s])`, tt.formatLiteral)
+			literalVal, err := parseAndEval(expr, t)
+			if err != nil {
+				t.Fatalf("program.Eval failed: %v", err)
+			}
+			out, err := parseAndEval(literalVal.Value().(string), t)
```

Comment:
Also validate that `literalVal.Value().(string) + " == " + formatLiteral?

Reply:
Fixed - I've added the equality check expression.